### PR TITLE
chore: add docs for network restrictions

### PR DESCRIPTION
**Pull request by darora**

### CODE REVIEW

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+export const meta = {
+  id: 'network-restrictions',
+  title: 'Network Restrictions',
+  description: "Apply network restrictions for your project's database.",
+}
+
+<Admonition type="note">
+
+Network Restrictions are currently in beta, and are being slowly made available to all projects.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:11]:
```suggestion
Network Restrictions are currently in beta and are being slowly made available to all projects.
```

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+The current iteration of Network Restrictions applies to connections to Postgres and Pgbouncer. Support for restricting connections to Postgrest, Storage, and Auth will be added at a later date.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:13]:
```suggestion
The current iteration of Network Restrictions applies to connections to Postgres and PgBouncer. Support for restricting connections to PostgREST, Storage, and Auth will be added at a later date.
```

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+</Admonition>
+
+Each Supabase project comes with configurable restrictions on the IP ranges that are allowed to connect to Postgres and Pgbouncer ("your database"). These restrictions are enforced before traffic reaches your database; if a connection is allowed by a network restriction, it will still need to authenticate successfully with valid credentials.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:17]:
```suggestion
Each Supabase project comes with configurable restrictions on the IP ranges that are allowed to connect to Postgres and PgBouncer ("your database"). These restrictions are enforced before traffic reaches your database. If a connection is not restricted by IP, it still needs to authenticate successfully with valid database credentials.
```

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+To get started:
+
+1. [Install](/docs/guides/cli) the Supabase CLI 1.22.0+.
+1. [Log in](/docs/guides/cli/local-development#log-in-to-the-supabase-cli) to your Supabase account using the CLI.
+1. Ensure that you have [Owner or Admin permissions](/docs/guides/hosting/platform#manage-team-members) for the project you'd like to set up network restrictions for.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:23]:
```suggestion
1. Ensure that you have [Owner or Admin permissions](/docs/guides/hosting/platform#manage-team-members) for the project that you are enabling network restrictions.
```

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+## Check restrictions
+
+You can use the `get` subcommand of the CLI to retrieve the restrictions currently in effect.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:27]: add link to `get`?

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+```bash
+> supabase network-restrictions --project-ref {ref} update --db-allow-cidr 0.0.0.0/0 --experimental
+DB Allowed CIDRs: [0.0.0.0/0]
+Restrictions applied successfully: true
+```
+
+## Limitations
+
+1. Network Restrictions do not currently apply to APIs offered over HTTPS (e.g. Postgrest, Storage, Auth).
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:70]:
```suggestion
1. Network Restrictions do not currently apply to APIs offered over HTTPS (e.g. PostgREST, Storage, Auth).
```

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+1. Network Restrictions should not be used if you require your edge functions to be able to connect to your project's postgres or pgbouncer endpoints.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:71]:
```suggestion
1. Network Restrictions should not be used if you require you need to connect to your Postgres database using Edge Functions.
```

**Review by kiwicopple — APPROVED** [review]

**Comment by MildTomato** [comment]
updated to new site :D added to new side menu constants

File: apps/docs/pages/guides/platform/network-restrictions.mdx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+> supabase network-restrictions --project-ref {ref} update --db-allow-cidr 0.0.0.0/0 --experimental
+DB Allowed CIDRs: [0.0.0.0/0]
+Restrictions applied successfully: true
+```
+
+## Limitations
+
+1. Network Restrictions do not currently apply to APIs offered over HTTPS (e.g. PostgREST, Storage, Auth).
+1. Network Restrictions should not be used if you require you need to connect to your Postgres database using Edge Functions.
```
Comment (dannykng) [inline apps/docs/pages/guides/platform/network-restrictions.mdx:71]:
```suggestion
1. Network Restrictions should not be used if you require you need to connect to your Postgres database using Edge Functions.

export const Page = ({ children }) => <Layout meta={meta} children={children} />

export default Page
```

**Pull request merged**